splink/splink_dataframe.py

```diff
@@ -55,13 +58,19 @@ def _check_drop_table_created_by_splink(self, force_non_splink_table=False):
             f"physical name {self.physical_name}"
         )
 
-    def drop_table_from_database(self, force_non_splink_table=False):
+    def _drop_table_from_database(self, force_non_splink_table=False):
         raise NotImplementedError(
-            "Drop table from database not implemented for this linker"
+            "_drop_table_from_database from database not " "implemented for this linker"
         )
 
+    def drop_table_from_database_and_remove_from_cache(
+        self, force_non_splink_table=False
+    ):
+        self.linker._remove_splinkdataframe_from_cache(self)
+        self._drop_table_from_database(force_non_splink_table=force_non_splink_table)
```

Comment:
Do we want to swap the order of these two, just in case `_drop_table_from_db` fails?

I don't really see why it would for our current linkers, but it may be slightly more robust to flip them.

`_remove_splinkdataframe_from_cache` just pops our cache, so that should be relatively sound across the board.

Reply:
Yes - good point, have done it.  No reason not to.